### PR TITLE
(delphi) highlight hexadecimal, octal, and binary numbers

### LINKED ISSUE #2368: Patch: Support hexadeciaml, octal, binary numbers in Delphi language
Hi!

I hope, syntax is fine, as I couldn't test it. (I hacked my delphi.min.js directly.)

````
--- delphi-original.js	Mon Jan 20 05:56:52 2020
+++ delphi.js	Sun Jan 26 20:56:09 2020
@@ -63,10 +63,31 @@ function(hljs) {
     keywords: KEYWORDS,
     illegal: /"|\$[G-Zg-z]|\/\*|<\/|\|/,
     contains: [
       STRING, CHAR_STRING,
       hljs.NUMBER_MODE,
+      {
+        // https://www.freepascal.org/docs-html/ref/refse6.html
+        className: 'numbers',
+        variants: [
+          {
+            // hexadecimal notation, e.g., $7F
+            begin: '\\$[0-9A-Fa-f]+',
+            relevance: 0
+          },
+          {
+            // octal notation, e.g., &42
+            begin: '\\&[0-7]+',
+            relevance: 0
+          },
+          {
+            // binary notation, e.g., %1010
+            begin: '\\%[01]+',
+            relevance: 0
+          }
+        ]
+      },
       CLASS,
       FUNCTION,
       DIRECTIVE
     ].concat(COMMENT_MODES)
   }; 
````

Cheers,
Robert

### PR DESCRIPTION
**Pull request by bttrx**
Hi Josh,

as recommended from you in #2368, this is my first PR. Please check. Thanks.

Source: https://www.freepascal.org/docs-html/ref/refse6.html

Cheers,
Robert

---

Closes #2368.

### CODE REVIEW

File: src/languages/delphi.js [hunk 1]
```diff
@@ -65,6 +65,27 @@ function(hljs) {
     contains: [
       STRING, CHAR_STRING,
       hljs.NUMBER_MODE,
+      {
```
Comment (yyyc514) [inline src/languages/delphi.js]: Can we move this into a variable like all the other rules?  ie, `NUMBER`.

File: src/languages/delphi.js [hunk 2]
```diff
@@ -65,6 +65,27 @@ function(hljs) {
...
+            relevance: 0
+          },
+          {
+            // Octal notation, e.g., &42.
+            begin: '&[0-7]+',
+            relevance: 0
+          },
+          {
+            // Binary notation, e.g., %1010.
+            begin: '%[01]+',
```
Comment (yyyc514) [inline src/languages/delphi.js]: Looks much better without the needless escaping. :)

**Comment by yyyc514** [comment]
Also, still needs a mention in the changelog. :)

**Comment by yyyc514** [comment]
I cleaned up a bit, if you'll just add a CHANGELOG entry now this should be good to go!

**Comment by yyyc514** [comment]
@bttrx Thanks for your contribution!

**Comment by bttrx** [comment]
@yyyc514 Thanks, Josh, for improving my PR.

**Pull request merged**